src/sql/src/plan/statement/ddl.rs

```diff
@@ -1316,17 +1310,86 @@ fn kafka_sink_builder(
                 ccsr_config,
             }
         }
-        Some(Format::Json) => {
-            if consistency_topic.is_some() {
-                // todo@jldlaughlin: fix this!
-                unsupported!("JSON sink with consistency topic")
-            }
-            KafkaSinkFormat::Json
-        }
+        Some(Format::Json) => KafkaSinkFormat::Json,
         Some(format) => unsupported!(format!("sink format {:?}", format)),
         None => unsupported!("sink without format"),
     };
 
+    let (consistency_topic, consistency_format) = match consistency {
+        Some(KafkaConsistency {
+            topic,
+            topic_format,
+        }) => match topic_format {
+            Some(Format::Avro(AvroSchema::CsrUrl {
+                url,
+                seed,
+                with_options,
+            })) => {
+                if seed.is_some() {
+                    bail!("SEED option does not make sense with sinks");
+                }
+                let schema_registry_url = url.parse::<Url>()?;
+                let ccsr_with_options = normalize::options(&with_options);
+                let ccsr_config = kafka_util::generate_ccsr_client_config(
+                    schema_registry_url.clone(),
+                    &config_options,
+                    ccsr_with_options,
+                )?;
+
+                (
+                    Some(topic),
+                    Some(KafkaSinkFormat::Avro {
+                        schema_registry_url,
+                        key_schema: None,
+                        value_schema: avro::get_debezium_transaction_schema().canonical_form(),
+                        ccsr_config,
+                    }),
+                )
+            }
+            Some(other) => unsupported!(format!("consistency format {}", &other)),
+            None => unsupported!("consistency topic without format"),
```

Comment:
I think this message should somehow say that  `CONSISTENCY FORMAT` is expected, as the statement already has a `FORMAT` in it.

To make it even easier for the user, can the `FORMAT` be used as `CONSISTENCY FORMAT` unless overridden?

Reply:
Great suggestion, thank you! I've done this:
> To make it even easier for the user, can the FORMAT be used as CONSISTENCY FORMAT unless overridden?
And added [a test here](https://github.com/MaterializeInc/materialize/pull/7608/commits/d90aad3574122cffdb3326d1ebb4af3dbd88c266#diff-37787161b8fbbc6abdba42d5590532844a08e9a73c4f5304a3228375af7b75bdR230).